nbconvert/preprocessors/tagremove.py

```diff
@@ -0,0 +1,101 @@
+"""
+Module containing a preprocessor that removes cells if they match
+one or more regular expression.
+"""
+
+# Copyright (c) IPython Development Team.
+# Distributed under the terms of the Modified BSD License.
+
+from traitlets import List, Unicode
+from . import ClearOutputPreprocessor
+
+
+class TagRemovePreprocessor(ClearOutputPreprocessor):
+    """
+    Removes cells from a notebook that have tags that designate they are to be
+    removed prior to exporting the notebook.
+
+    Traitlets:
+    ----------
+    remove_cell_tags: removes cells tagged with these values
+    remove_all_output_tags: removes entire output areas on cells
+                            tagged with these values
+    remove_single_output_tags: removes individual output objects on
+                               outputs tagged with these values
+
+    """
+
+    remove_cell_tags = List(Unicode, default_value=[]).tag(config=True)
+    remove_all_outputs_tags = List(Unicode, default_value=[]).tag(config=True)
+    remove_single_output_tags = List(Unicode, default_value=[]).tag(config=True)
```

Comment:
Since the order doesn't matter here, I think you could make them sets instead of lists. That means we can use set operations to check if any tag matches:

```python
self.remove_cell_tags.intersection(cell.get('metadata', {}).get('tags', []))
```